crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -325,7 +332,7 @@ impl<TYPES: NodeType> Inner<TYPES> {
 
                                 return Ok(());
```

Comment:
If we don't return here we could get away without the outer loop I mentioned in my other comment.  I also think we want to use an `LruCatch` here just like in the `LatestViewSyncCertificate` case

Reply:
Should we also consider actually having the cache a layer up? This way it can be shared for every task instead of just per task. The deduplication in this case was just so we don't pull multiple of the same latest proposal

Reply:
Not sure I understand what you mean by per task.  If you mean the task we spawn to poll the webserver then yeah we should probably do a shared cache between the view number based and latest polls.  On the other hand we could probably just get rid of the view based polling for proposals after this.  I say we keep it as is for now

Reply:
Yeah, the cache is intra-task and not inter-task